src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1627,6 +1630,43 @@ void TranslateToFuzzReader::fixAfterChanges(Function* func) {
         i--;
       }
     }
+
+    void visitRethrow(Rethrow* curr) {
+      if (!isValidRethrow(curr->target)) {
+        replace();
+      }
+    }
+
+    bool isValidRethrow(Name target) {
+      // The rethrow must be on top.
+      assert(!expressionStack.empty());
+      assert(expressionStack.back()->is<Rethrow>());
+      if (expressionStack.size() < 2) {
+        // There must be a try for this rethrow to be valid.
+        return false;
+      }
+      Index i = expressionStack.size() - 2;
+      while (1) {
+        auto* curr = expressionStack[i];
+        if (auto* tryy = curr->dynCast<Try>()) {
+          // The rethrow must target a try, and must be nested in a catch of
+          // that try (not the body). Look at the child above us to check, when
+          // we find the proper try.
+          if (tryy->name == target) {
+            if (i + 1 >= expressionStack.size()) {
+              return false;
+            }
+            auto* child = expressionStack[i + 1];
+            return child != tryy->body;
+          }
+        }
+        if (i == 0) {
+          // We never found our try.
+          return false;
+        }
+        i--;
```

Comment:
Can we remove this if we change the `while (1)` above to
```cpp
for (Index i = expressionStack.size() - 2; i > 0; i--)
```
?

Reply:
Hmm, but we do need to process the item with index 0. The try might be the top-level item, with that index.

It is kind of tricky to make a for-style loop handle this given the unsigned indexing... I think unrolling it out into a while is best, but I'm open to other ideas.

Reply:
Ah right. Nevermind.